### ISSUE TITLE
A lot of bugs and exceptions when cloning

### ISSUE THREAD

**Issue by FauxKiwi** [kotlin-graphics/imgui#160]
I cloned the repository into IntelliJ and ran it.
At first it seemed to work well, but then I ran into a lot of things that didn't work.
To enumerate those bugs I found:
- Dropdown menus not working: StringFormatException or something 
- Slider for float and double not working: NumberFormatException
- Slider for colors not working: NumberFormatException as well
- The resizeCallback menu: NullPointerException
- Slider for UInt and other unsigned types: Don't remember
- Drag to copy: Also don't remember
And: The font menu in the style menu isn't working.
To summarize it just a bit
How can I fix that issues?

**Comment by elect86** [comment]
Thanks for reporting, 
you can clone the repo and PR some fixes, otherwise I'll go through the list in the next days

**Comment by FauxKiwi** [comment]
I fixed the first two bugs and created a pull request.
Perhaps what I did is not intended, so it's just my approach.
And for the first one I actually just commented out the problematic byte. So I think there are better solutions for that one.
Apart from that I have one question: How do I change the font?

**Comment by elect86** [comment]
> 
> 
> I fixed the first two bugs and created a pull request.

Thanks

> Perhaps what I did is not intended, so it's just my approach.

It's absolutely fine

> And for the first one I actually just commented out the problematic byte. So I think there are better solutions for that one.

We can stay like this for the moment till some other has a better solution or some other problems rise

> Apart from that I have one question: How do I change the font?

[`ImGui.io.fonts.addFontFromFileTTF("fonts/ArialUni.ttf", 16f, glyphRanges = imgui.font.glyphRanges.japanese)`](https://github.com/kotlin-graphics/imgui/blob/master/gl/src/test/kotlin/examples/opengl3.kt#L125)

**Comment by FauxKiwi** [comment]
I have a question:
Do you have these bugs, too?

**Comment by FauxKiwi** [comment]
@elect86 Do you have these bugs, too?

**Comment by elect86** [comment]
> 
> 
> @elect86 Do you have these bugs, too?

Going through.

I already fixed a couple regarding typing the colorEdit widgets

What do you mean by "Dropdown menus"?

**Comment by FauxKiwi** [comment]
I fixed the bug regarding the dropdown menus, too (e.g. the "examples" tab in the menu bar).
I think I fixed all bugs with the sliders.
Also, there is one bug I forgot: The Font tab in the style editor crashes the program. But I think I can fix this

**Comment by elect86** [comment]
Unsigned sliders are also fixed

**Comment by FauxKiwi** [comment]
Okay nice

**Comment by FauxKiwi** [comment]
Could you please fix these bugs in the docking branch as well

**Comment by elect86** [comment]
yeah, I'll merge them back there

**Comment by elect86** [comment]
Edit: it looks like I've already done this yesterday, have you pulled?

**Comment by FauxKiwi** [comment]
I pulled it now and it's working
Is it possible to make the windows undecorated when using viewports?

**Comment by elect86** [comment]
yeah, that's a bug I still have to fix..

**Comment by FauxKiwi** [comment]
I think there are lots of bugs with docking and viewports.
How do I dock a window without using the dockspace of the demo-window?

**Comment by elect86** [comment]
> 
> 
> I think there are lots of bugs with docking and viewports.
> How do I dock a window without using the dockspace of the demo-window?

Hi @FauxKiwi,

sorry for the delay, I was kept busy by other stuff

Can you list me some bugs and elaborate what did you mean exactly by that?

**Comment by FauxKiwi** [comment]
- Normally, you can't even open the dockspace: NoSuchElemtException: List is empty. This is related to the menu bar. When I comment out the menu bar, there are more bugs:
- You can't resize docked windows
- When you docked a window and are using an imgui.ini, restarting the program throws an NumberFormatException
- When you dock one window onto another, they begin to move (?)

**Comment by elect86** [comment]
> 
> 
>     * Normally, you can't even open the dockspace: NoSuchElemtException: List is empty. This is related to the menu bar. When I comment out the menu bar, there are more bugs:
> 
>     * You can't resize docked windows
> 
>     * When you docked a window and are using an imgui.ini, restarting the program throws an NumberFormatException
> 
>     * When you dock one window onto another, they begin to move (?)


Table branch has been finally merged on the original repo, I need to finish to sync that first so that it doesn't lag much behind, then I'll come back to these

**Comment by elect86** [comment]
Also, if you want step in and try to fix them yourself, do not hesitate

I'll offer you a pizza at the end :d

**Comment by FauxKiwi** [comment]
Also, there is a bug with imgui.ini: NumberFormatException: For input string: "Space ID=0x3bc79352 Window=0x4647b76e Pos=0" <br>
I will see what bugs I can fix but I'm not so much into the project than you are. So often I don't exactly know the problem

**Comment by elect86** [comment]
which branch?

**Comment by FauxKiwi** [comment]
docking, when you docked a window
